omegaconf/dictconfig.py

```diff
@@ -292,9 +292,7 @@ def _s_validate_and_normalize_key(self, key_type: Any, key: Any) -> DictKeyType:
             return key  # type: ignore
         elif issubclass(key_type, Enum):
             try:
-                ret = EnumNode.validate_and_convert_to_enum(
-                    key_type, key, allow_none=False
-                )
+                ret = EnumNode.validate_and_convert_to_enum(key_type, key)
```

Comment:
It looks like this PR has modified the `EnumNode.validate_and_convert_to_enum` function, so that it no longer accepts an `allow_none` argument. The new version of the function behaves as if `allow_none=False` had been passed in.

Reply:
There is a case in `test_errors.py` that covers `__setitem__(cfg, None, ...)` for Enum key_type:
```python
    pytest.param(
        Expected(
            create=lambda: DictConfig(key_type=Color, element_type=str, content={}),
            op=lambda cfg: cfg.__setitem__(None, "bar"),
            exception_type=KeyValidationError,
            msg="Key 'None' is incompatible with the enum type 'Color', valid: [RED, GREEN, BLUE]",
            key=None,
        ),
        id="DictConfig[Color,str]:setitem_bad_key",
    ),
```

There are also a few tests that cover this code path indirectly, e.g. in `test_basic_ops_dict.py` there is the `test_in_dict` function which covers the case of `assert (None in cfg_with_Enum_keys) == False`.

Do you think we should be testing the behavior of `_s_validate_and_normalize_key` directly?

Reply:
> It looks like this PR has modified the `EnumNode.validate_and_convert_to_enum` function, so that it no longer accepts an `allow_none` argument. The new version of the function behaves as if `allow_none=False` had been passed in.

Yes that is correct, there is no need to pass `allow_none=False` because it is now the behavior by default (this was intended).